test/txRelay.js

```diff
@@ -315,18 +323,125 @@ contract('TxRelay', (accounts) => {
 
         assert.equal(p.nonce, i.toString(), "Nonce should have updated")
-        await txRelay.relayMetaTx(p.v, p.r, p.s, p.dest, p.data, {from: sender})
+        await txRelay.relayMetaTx(p.v, p.r, p.s, p.dest, p.data, 0, {from: sender})
 
         regData = await mTestReg.registry.call(user1)
         assert.equal(regData.toNumber(), randNum, 'Registry did not update properly')
       }
     }).timeout(10000000)
   })
 
+  describe("Whitelist", () => {
+    let sender1 = accounts[0]
+    let sender2 = accounts[1]
+    let sender3 = accounts[2]
+    let sender4 = accounts[3]
+    let sender5 = accounts[4]
+    let sender6 = accounts[5]
+    let sender7 = accounts[6]
+    before(async () => {
+      testTxRelay = await MetaTxRelay.new()
+
+    })
+
+    it("Should register a list of addresses to a whitelist", async () => {
+      await testTxRelay.addToWhitelist([sender2, sender3], {from: sender1})
+
+      assert.isTrue(await testTxRelay.whitelist.call(sender1, sender2), "sender2 should be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(sender1, sender3), "sender3 should be on list")
+
+      assert.isFalse(await txRelay.whitelist.call(sender1, sender1), "sender1 should not be on list")
+    })
+
+    it("Should allow owner of list to add another sender", async () => {
+      await testTxRelay.addToWhitelist([sender4], {from: sender1})
+
+      assert.isTrue(await testTxRelay.whitelist.call(sender1, sender2), "sender2 should be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(sender1, sender3), "sender3 should be on list")
+      assert.isTrue(await testTxRelay.whitelist.call(sender1, sender4), "sender4 should be on list")
+    })
+
+    it("Should not allow non-owner to add another sender", async () => {
+      await testTxRelay.addToWhitelist([sender5], {from: sender2})
+
+      assert.isFalse(await txRelay.whitelist.call(sender1, sender5), "sender5 should not be on list")
+    })
+
+    it("Should allow owner of list to remove senders", async () => {
+      await testTxRelay.removeFromWhitelist([sender4, sender3], {from: sender1})
+
+      assert.isTrue(await testTxRelay.whitelist.call(sender1, sender2), "sender2 should be on list")
+      assert.isFalse(await testTxRelay.whitelist.call(sender1, sender3), "sender3 not should be on list")
+      assert.isFalse(await testTxRelay.whitelist.call(sender1, sender4), "sender4 not should be on list")
+    })
+
+    it("Removing sender not on list should not have an effect", async () => {
+      assert.isFalse(await testTxRelay.whitelist.call(sender1, sender3), "sender3 not should be on list")
+
+      await testTxRelay.removeFromWhitelist([sender3], {from: sender1})
+
+      assert.isFalse(await testTxRelay.whitelist.call(sender1, sender3), "sender3 not should be on list")
+    })
+
+    it("Should allow another sender to create their own whitelist", async () => {
+      await testTxRelay.addToWhitelist([sender5, sender6], {from: sender2})
```

Comment:
`sender5` was actually added on line 369. Just saying.

Reply:
Yep, I know. Shouldn't affect the test though.